aliasmgr.go

```diff
@@ -0,0 +1,446 @@
+package lnd
+
+import (
+	"encoding/binary"
+	"fmt"
+	"sync"
+
+	"github.com/lightningnetwork/lnd/htlcswitch/hop"
+	"github.com/lightningnetwork/lnd/kvdb"
+	"github.com/lightningnetwork/lnd/lnwire"
+)
+
+var (
+	// aliasBucket stores aliases as keys and their base SCIDs as values.
+	// This is used to populate the maps that the aliasMgr uses. The keys
+	// are alias SCIDs and the values are their respective base SCIDs. This
+	// is used instead of the other way around (base -> alias...) because
+	// updating an alias would require fetching all the existing aliases,
+	// adding another one, and then flushing the write to disk. This is
+	// inefficient compared to N 1:1 mappings at the cost of marginally
+	// more disk space.
+	aliasBucket = []byte("alias-bucket")
+
+	// confirmedBucket stores whether or not a given base SCID should no
+	// longer have entries in the ToBase maps. The key is the SCID that is
+	// confirmed with 6 confirmations and is public, and the value is
+	// empty.
+	confirmedBucket = []byte("base-bucket")
+
+	// aliasAllocBucket is a root-level bucket that stores the last alias
+	// that was allocated. It is used to allocate a new alias when
+	// requested.
+	aliasAllocBucket = []byte("alias-alloc-bucket")
+
+	// lastAliasKey is a key in the aliasAllocBucket whose value is the
+	// last allocated alias ShortChannelID. This will be updated upon calls
+	// to requestAlias.
+	lastAliasKey = []byte("last-alias-key")
+
+	// invoiceAliasBucket is a root-level bucket that stores the alias
+	// SCIDs that our peers send us in the funding_locked TLV. The keys are
+	// the ChannelID generated from the FundingOutpoint and the values are
+	// the remote peer's alias SCID.
+	invoiceAliasBucket = []byte("invoice-alias-bucket")
+
+	// byteOrder denotes the byte order of database (de)-serialization
+	// operations.
+	byteOrder = binary.BigEndian
+
+	// startingAlias is the first alias ShortChannelID that will get
+	// assigned by requestAlias. The starting BlockHeight is chosen so that
+	// legitimate SCIDs in integration tests aren't mistaken for an alias.
+	startingAlias = lnwire.ShortChannelID{
+		BlockHeight: 10000,
+		TxIndex:     0,
+		TxPosition:  0,
+	}
+
+	// errNoBase is returned when a base SCID isn't found.
+	errNoBase = fmt.Errorf("no base found")
+
+	// errNoPeerAlias is returned when the peer's alias for a given
+	// channel is not found.
+	errNoPeerAlias = fmt.Errorf("no peer alias found")
+)
+
```

Comment:
i don't see a clear win from changing the db structure or internal maps. i do agree some of the naming can change and an api seems fine

Reply:
even with a naming/api change which im not against, i don't see what is functionally different though - maybe i don't understand the suggestion?

the gossiper:
- must use special gossip maps for lookup to fetch an SCID. this is then used to lookup the edge in the graph

the fmgr:
- must "notify" the aliasmgr to no longer persist the gossip mappings since the channel is public with 6 confs
- must insert aliases into the main alias map
  - must also optionally insert into a separate map used by the gossiper depending on whether this is an upgrade case or not.

the peer:
- must insert aliases into the main alias map in upgrade cases, and bypass the gossip map

the switch:
- must use the main alias map for forwarding and failing back channel updates